### PR TITLE
Animation WIP

### PR DESCRIPTION
**Pull request by mbr4nt**
leaving it open for review

### CODE REVIEW

File: AwePointsAnimation.cm [hunk 1]
```diff
@@ -109,9 +109,10 @@ public class AwePointsAnimation extends AweAnimation {
     }
 
     extend public void setViewMode(ViewMode viewMode) {
-        if (World w = session.?mainWorld) {
-            w.setViewMode(viewMode);
-        }
+        // if (World w = session.?mainWorld) {
+        //     w.setViewMode(viewMode);
+        // }
```
Comment (mbr4nt) [inline AwePointsAnimation.cm]: this should probably go to the parent class, AweAnimation

File: AwePointsAnimation.cm [hunk 2]
```diff
@@ -185,6 +182,10 @@ public class AwePointsAnimation extends AweAnimation {
         if (space) {
             space.beginAnimateDrawSequence(this);
             space.show(this);
+
```
Comment (mbr4nt) [inline AwePointsAnimation.cm]: could we move this visibility stuff inside the class and make the constructor parameterless

File: AwePointsAnimation.cm [hunk 3]
```diff
@@ -185,6 +182,10 @@ public class AwePointsAnimation extends AweAnimation {
...
+            ViewMode vm = AweElectricalViewMode( visible={#normal, #cad} );
+            this.space._viewMode = vm;
+            this.space.viewModeChanged();
```
Comment (mbr4nt) [inline AwePointsAnimation.cm]: any reason not to use the space.setViewMode method ?

File: electrical/AweElectricalAnimation.cm [hunk 4]
```diff
@@ -78,12 +78,21 @@ public class AweElectricalAnimation extends AwePointsListAnimation {
...
         Awe3D prims();
-        prims << status.?get3D();
+
         prims << currentMode.?get3D();
         return prims;
     }
 
+    public void refresh() {
+        ViewMode vm = session.?mainWorld.viewMode("electrical");
+        if ( vm ) {
```
Comment (mbr4nt) [inline electrical/AweElectricalAnimation.cm]: you have a pointer to it (this.viewMode)

File: AwePointsAnimation.cm [hunk 5]
```diff
@@ -109,9 +109,10 @@ public class AwePointsAnimation extends AweAnimation {
     }
 
     extend public void setViewMode(ViewMode viewMode) {
-        if (World w = session.?mainWorld) {
-            w.setViewMode(viewMode);
-        }
+        // if (World w = session.?mainWorld) {
```
Comment (mbr4nt) [inline AwePointsAnimation.cm]: keep a pointer to the view

File: AwePointsAnimation.cm (hunk 3 above)
Comment (twlamb) [inline AwePointsAnimation.cm]: Yeah I read the internals of that method wrong, that should work.

**Comment by twlamb** [comment]
#156 includes all of these changes, so I'm closing.

**Pull request closed without merging**